src/components/DarkModeToggle/__tests__/index.test.tsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import DarkModeToggle from '../index';
+
+let mockCurrentTheme = '';
+const mockToggleTheme = jest.fn().mockImplementation((newTheme: string) => {
+  mockCurrentTheme = newTheme === 'light' ? 'dark' : 'light';
+});
+
+// mock dark mode module for controlling dark mode HOC behavior
+jest.mock('@skagami/gatsby-plugin-dark-mode', () => ({
+  __esModule: true,
+  useTheme: () => [mockCurrentTheme, mockToggleTheme],
+}));
+
+describe('DarkModeToggle Component', () => {
+  it('render dark mode toggle', () => {
+    const { container } = render(<DarkModeToggle />);
+    expect(container).toMatchSnapshot();
+  });
+
+  it('switches dark theme to light theme', async () => {
+    mockCurrentTheme = 'dark';
+    render(<DarkModeToggle />);
+    const toggle = await screen.findByText('Toggle Dark Mode');
+    await userEvent.click(toggle);
+  });
+
+  it('switches light theme to dark theme', async () => {
+    mockCurrentTheme = 'light';
+    render(<DarkModeToggle />);
+    const toggle = await screen.findByText('Toggle Dark Mode');
+    await userEvent.click(toggle);
```

Comment:
I think your tests need to have an `expect()` to check if the resulting `mockCurrentTheme` is correct.

E.g.:

```jsx
  it('switches light theme to dark theme', async () => {
    mockCurrentTheme = 'light';
    render(<DarkModeToggle />);
    const toggle = await screen.findByText('Toggle Dark Mode');
    await userEvent.click(toggle);
    expect(mockCurrentTheme).toBe('dark');
  });

Reply:
Hi @ovflowd thank you for the suggestion. I have modified my tests accordingly to check the currentTheme after toggling. Also calling it out in advance, while I rebased my branch with main, I had some merge conflicts with firebase.json (it just gets pulled and updated everytime I try to rebase) and I have resolved it to match it with main. Just letting you know that it was not something I intended to change as part of this commit. Looking forward for your feedback.